src/main/resources/include/hail/NativeStatus.h

```diff
@@ -0,0 +1,47 @@
+#ifndef HAIL_NATIVESTATUS_H
+#define HAIL_NATIVESTATUS_H 1
+
+#include "hail/NativeObj.h"
+#include <cstdarg>
+#include <cstdio>
+#include <string>
+
+namespace hail {
+
+class NativeStatus : public NativeObj {
+public:
+  int errno_;
+  std::string msg_;
+  std::string location_;
+  
+public:
+  inline NativeStatus() : errno_(0) { }
+  
+  virtual ~NativeStatus() { }
+  
+  inline void clear() {
+    // When errno_ == 0, the values of msg_ and location_ are ignored
+    errno_ = 0;
+  }
+  
+  void set(const char* file, int line, int code, const char* msg, ...) {
+    char buf[8*1024];
```

Comment:
This is a ticking time bomb.  I think we should standardize on the fmt library for building strings: https://github.com/fmtlib/fmt.